dids/src/main/kotlin/web5/sdk/dids/DidWeb.kt

```diff
@@ -0,0 +1,127 @@
+ *
+ * @property uri The URI of the "did:web" which conforms to the DID standard.
+ * @property keyManager A [KeyManager] instance utilized to manage the cryptographic keys associated with the DID.
+ *
+ * ### Usage Example:
+ * ```kotlin
+ * val keyManager = InMemoryKeyManager()
+ * val did = StatefulWebDid("did:web:tbd.website", keyManager)
+ * ```
+ */
+public class DidWeb(
+  uri: String,
+  keyManager: KeyManager
+) : Did(uri, keyManager)
+
+/**
+ * Configuration options for the [DidWebApi].
+ *
+ * - [engine] is used to override the default ktor engine, which is [CIO].
+ */
+public class DidWebApiConfiguration internal constructor(
+  public var engine: HttpClientEngine? = CIO.create { },
+)
+
+/**
+ * Returns a [DidWebApi] instance after applying [blockConfiguration].
+ */
+public fun DidWebApi(blockConfiguration: DidWebApiConfiguration.() -> Unit): DidWebApi {
+  val conf = DidWebApiConfiguration().apply(blockConfiguration)
+  return DidWebApiImpl(conf)
+}
+
+private class DidWebApiImpl(configuration: DidWebApiConfiguration) : DidWebApi(configuration)
+
+private const val wellKnownURLPath = "/.well-known"
+private const val didDocFilename = "/did.json"
+
+/**
+ * Implements [resolve] and [create] according to https://w3c-ccg.github.io/did-method-web/
+ */
+public sealed class DidWebApi(
+  configuration: DidWebApiConfiguration
+) : DidMethod<DidWeb, CreateDidOptions> {
+
+  private val mapper = jacksonObjectMapper()
+
+  private val engine: HttpClientEngine = configuration.engine ?: CIO.create {}
+
+  private val client = HttpClient(engine) {
+    install(ContentNegotiation) {
+      jackson { mapper }
+    }
+  }
+
+  override val methodName: String = "web"
+
+  override fun resolve(did: String, options: ResolveDidOptions?): DidResolutionResult {
+    val docURL = getDocURL(did)
+
+    val resp = runBlocking {
+      client.get(docURL) {
+        contentType(ContentType.Application.Json)
+      }
+    }
+
+    val body = runBlocking { resp.bodyAsText() }
+
+    if (!resp.status.isSuccess()) {
+      throw InvalidStatusException(resp.status.value, "resolution error response: '$body'")
+    }
+    return DidResolutionResult(
+      didDocument = mapper.readValue(body, DIDDocument::class.java),
+    )
+  }
+
+  private fun getDocURL(didWebStr: String): String {
+    val parsedDid = DID.fromString(didWebStr)
+    require(parsedDid.methodName == methodName) {
+      "$didWebStr is missing prefix \"did:$methodName\""
+    }
+
+    val domainNameWithPath = parsedDid.methodSpecificId.replace(":", "/")
+    val decodedDomain = URLDecoder.decode(domainNameWithPath, UTF_8)
+
+    val targetUrl = StringBuilder("https://$decodedDomain")
+
+    val url = URL(targetUrl.toString())
+    if (url.path.isEmpty()) {
+      targetUrl.append(wellKnownURLPath)
+    }
+    targetUrl.append(didDocFilename)
+    return targetUrl.toString()
+  }
+
+  public override fun create(keyManager: KeyManager, options: CreateDidOptions?): DidWeb {
```

Comment:
I think we should implement this so that users can create a did web doc similar to the way ssi service does it:

curl -X PUT https://ssi.tbddev.org/v1/dids/web \
-H "Content-Type: application/json" \
-d '{
  "keyType": "Ed25519",
  "options": {
    "didWebId": "did:web:example123.com"
  }
}'

Reply:
I agree but think this sdk should be decoupled from a future http api